### PR TITLE
Mongo7 upgrade

### PR DESCRIPTION
**Pull request by Xiangs18**

### CODE REVIEW

**Comment by Xiangs18** [comment]
RetryWrites will be included in the next PR.

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -22,13 +22,12 @@ jobs:
       fail-fast: false
       matrix:
         include:
-          # the current production setup
           - java: '8'
-            mongo: 'mongodb-linux-x86_64-3.6.13'
-            gradle_test: 'test'
+            mongo: 'mongodb-linux-x86_64-ubuntu2204-7.0.4'
+            wired_tiger: 'false'
```
Comment (MrCreosote) [inline .github/workflows/test.yml:27]: Wired tiger is the only option for non-enterprise MongoDB 4.2+

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -22,13 +22,12 @@ jobs:
...
           - java: '11'
             mongo: 'mongodb-linux-x86_64-3.6.23'
```
Comment (MrCreosote) [inline .github/workflows/test.yml:29]: Should change this to the actual version that's running in Spin

File: .github/workflows/test.yml (hunk 1 above)
Comment (MrCreosote) [inline .github/workflows/test.yml:27]: Wired tiger is the only option for non-enterprise MongoDB 4.2+

File: .github/workflows/test.yml (hunk 2 above)
Comment (MrCreosote) [reply .github/workflows/test.yml:29]: If the version in spin is 4.2+ there's no need for the wired tiger matrix entry at all. If it's lower than that, then the test config setup needs to be modified as well

**Review by MrCreosote — COMMENTED** [review]
Mostly looks good, just needs some tweaks for the test matrix

**Comment by MrCreosote** [comment]
Forgot about release notes, unless they'll go in the next version

File: .github/workflows/test.yml (hunk 2 above)
Comment (Xiangs18) [reply .github/workflows/test.yml:29]: Based on Jason's response, the image is mongo:4, which I assume is lower than version 4.2 and therefore requires switching to mongodb-linux-x86_64-4.0?

File: .github/workflows/test.yml (hunk 1 above)
Comment (Xiangs18) [reply .github/workflows/test.yml:27]: Are you saying wired_tiger should always be `true` for non-enterprise MongoDB 4.2+

Comment (MrCreosote) [reply .github/workflows/test.yml:27]: I'm saying it doesn't matter as it's the only choice. If both options are 4.2+ we can just ignore it

Comment (Xiangs18) [reply .github/workflows/test.yml]: 👍

File: .github/workflows/test.yml (hunk 2 above)
Comment (Xiangs18) [reply .github/workflows/test.yml:28]: 👍

**Comment by Xiangs18** [comment]
> Forgot about release notes, unless they'll go in the next version

next PR

File: .github/workflows/test.yml (hunk 2 above)
Comment (MrCreosote) [reply .github/workflows/test.yml:28]: To be clear, you need to get the mongo server version from Jason. It *might* be 4.0, we don't know based soley on the tag

Comment (Xiangs18) [reply .github/workflows/test.yml]: done

**Review by MrCreosote — APPROVED** [review]
LGTM

**Pull request merged**